alf/algorithms/config.py

```diff
@@ -38,6 +38,9 @@ def __init__(self,
                  eval_interval=10,
                  epsilon_greedy=0.,
                  eval_uncertainty=False,
+                 hold_out_dataset=None,
+                 train_classes=None,
+                 hold_out_classes=None,
```

Comment:
The easiest entry point to handle datasets is within the SL_trainer class in the policy_trainer file.
One compromise I see is to load the full datasets within the trainer, but handle what classes are used for training/testing within the algorithm. In that way, these options don't need to be part of config. 
But I think it would be odd to handle all dataset creation within the algorithm.

Reply:
The problem is that for supervised learning, the form of data can be quite different. I made a refactoring so that the algorithm will have an argument `data_creator` for creating data see PR #840.

Reply:
I see, this is good. I'll incorporate this data creator.